mlflow/pyfunc/__init__.py

```diff
@@ -840,7 +856,11 @@ def metadata(self):
     @property
     def model_config(self):
         """Model's flavor configuration"""
-        return self._model_meta.flavors[FLAVOR_NAME].get(MODEL_CONFIG, {})
+        model_config = self._model_meta.flavors[FLAVOR_NAME].get(MODEL_CONFIG, {})
+        if isinstance(model_config, str):
+            return _validate_and_get_model_config_from_file(model_config)
```

Comment:
@sunishsheth2009 Instead of storing the model config as a file, can we always load it as a dictionary when the config is passed to `mlflow.pyfunc.log_model()` and then save the dictionary in the flavor conf? This will remove a lot of special casing & if / else logic.

Reply:
Yes agreed. Based on the code as well when we load the model it should always be a dict for pyfunc usecase.

Reply:
I can do this - I originally thought we wanted to maintain dict -> dict and file -> file from log_model to the MLmodel file. But if this does not matter we can just have dict everywhere.

Reply:
The langchain implementation currently maintains file everywhere (the opposite), we may want to make this also have dict everywhere instead especially since it is also logging pyfunc flavor.